### PR TITLE
Transfer map API

### PR DESCRIPTION
**Pull request by thatcosmonaut**
Adds SDL_GpuMapTransferBuffer and SDL_GpuUnmapTransferBuffer. This allows the client to do things like build vertex data directly in the transfer buffer, which removes a copy step. After mapping the client must unmap before encoding any Upload commands or calling SetTransferData or GetTransferData.

### CODE REVIEW

File: include/SDL3/SDL_gpu.h [hunk 1]
```diff
@@ -1514,7 +1514,43 @@ extern DECLSPEC void SDLCALL SDL_GpuEndComputePass(
...
+ * Maps a transfer buffer into application address space.
+ * You must unmap the transfer buffer before encoding upload commands
+ * or calling SetTransferData or GetTransferData.
+ *
+ * \param device a GPU context
+ * \param transferBuffer a transfer buffer
+ * \param offsetInBytes offset from the beginning of the transfer buffer
+ * \param sizeInBytes number of bytes to map
+ * \param cycle if SDL_TRUE, cycles the transfer buffer if it is bound
+ * \param ppData a pointer to an application-accessible pointer
```
Comment (kg) [inline include/SDL3/SDL_gpu.h:1529]: would rephrase this like
`where to store the address of the mapped transfer buffer memory`

**Review by flibitijibibo — APPROVED** [review]
Pretty uninvasive feature, which is a nice change of pace!

Can merge when it gets approval from @kg and @TheSpydog

**Review by kg — APPROVED** [review]
As discussed on discord, I'm ok with removing set/get transfer data convenience functions for simplification, though I would like to see them stay.
We also discussed whether it makes sense to remove the offset and size arguments. I'm OK with removing them if you want to, but it seems fine to keep them.

File: src/gpu/metal/SDL_gpu_metal.m [hunk 2]
```diff
@@ -1439,6 +1439,7 @@ static void METAL_INTERNAL_CycleActiveBuffer(
 static SDL_GpuTransferBuffer* METAL_CreateTransferBuffer(
     SDL_GpuRenderer *driverData,
     SDL_GpuTransferUsage usage,
+    SDL_GpuTransferBufferMapFlags mapFlags,
```
Comment (flibitijibibo) [inline src/gpu/metal/SDL_gpu_metal.m:1442]:
```suggestion
    SDL_GpuTransferBufferMapFlags mapFlags, /* Ignored on Metal */
```

File: include/SDL3/SDL_gpu.h [hunk 3]
```diff
@@ -1514,7 +1524,38 @@ extern DECLSPEC void SDLCALL SDL_GpuEndComputePass(
...
+/* TransferBuffer Data */
+
+/**
+ * Maps a transfer buffer into application address space.
+ * You must unmap the transfer buffer before encoding upload commands.
+ *
+ * \param device a GPU context
+ * \param transferBuffer a transfer buffer
+ * \param cycle if SDL_TRUE, cycles the transfer buffer if it is bound
+ * \param ppData a pointer to an application-accessible pointer
```
Comment (flibitijibibo) [inline include/SDL3/SDL_gpu.h:1536]:
```suggestion
 * \param ppData where to store the address of the mapped transfer buffer memory
```
